src/formatting/lists.rs

```diff
@@ -741,22 +753,28 @@ where
                 .span_to_snippet(mk_sp(self.prev_span_end, (self.get_lo)(&item)))
                 .unwrap_or("");
             let (pre_comment, pre_comment_style) = extract_pre_comment(pre_snippet);
+            let next_item = self.inner.peek();
 
             // Post-comment
-            let next_start = match self.inner.peek() {
+            let next_start = match next_item {
                 Some(next_item) => (self.get_lo)(next_item),
                 None => self.next_span_start,
             };
             let post_snippet = self
                 .snippet_provider
                 .span_to_snippet(mk_sp((self.get_hi)(&item), next_start))
                 .unwrap_or("");
-            let comment_end = get_comment_end(
-                post_snippet,
-                self.separator,
-                self.terminator,
-                self.inner.peek().is_none(),
-            );
+            let comment_end =
+                if next_item.is_none() && last_line_contains_single_line_comment(post_snippet) {
```

Comment:
what condition is the comment check for? a small example snippet would be helpful, my mental runtime is failing me :smile:

Reply:
@calebcartwright The trailing comment of last item is not capturing for the item list. So I put this condition to take the trailing comment if it in the same line.

Without this condition:-

```rust
use c; // cpost
use b; // bpost
use a; // Trailing comment of last item
```
formatting to

```rust
use a;
use b; // bpost
use c; // cpost // Trailing comment of last item
```